### PR TITLE
Added openbsd aliases and platform detection

### PR DESCRIPTION
**Pull request by ddkn**
In response to issue #3354, here is a PR for the patch.

Cheers,

### CODE REVIEW

**Comment by laloch** [comment]
@ddkn, could you please provide a news item for the change? See [Developer’s Guide](https://xon.sh/devguide.html) for details.
@xonsh/xore, anyone able/willing to test on OpenBSD?

**Comment by laloch** [comment]
@ddkn, also please check CI results. Currently there are several Flake8 check failures.

**Comment by ddkn** [comment]
I was able to successfully install xonsh with my patches by issuing `python3 setup.py install --user --prefix=` and it installed and the aliases work fine. This is being tested on OpenBSD 6.5 stable.

**Comment by ddkn** [comment]
I am perplexed by the error generated by `ci/circleci: build_black`. I have only checked the files I have modified, since the only thing I have done is copy lines from master and change NetBSD|NETBSD|netbsd to OpenBSD|OPENBSD|openbsd and added bool(ON_OPENBSD) it complains about reformatting platform.py. I am unsure as to correct this as it looks the same to me. 

pylint wants corrections on lines I never touched.
```
fool$ pylint xonsh/platform.py xonsh/aliases.py 
No config file found, using default configuration
************* Module xonsh.platform
E:324, 0: invalid syntax (syntax-error)
************* Module xonsh.aliases
E: 92, 0: invalid syntax (syntax-error
```

xonsh/platform.py
```
        def is_dir(self, *, follow_symlinks=True):
```

xonsh/alias.py
```
            token, *rest = map(expand_path, value)
```

This is odd to me as I do not see how these errors got through the automated build system. Any advice on this?

**Comment by laloch** [comment]
Black basically complains about the same as Flake8 (PEP8) - the two missing empty lines around `def ON_BSD()`. I don't really understand why they didn't complain in the past. Just go ahead and add those two lines to make them happy.
Sorry for the inconvenience.

**Comment by laloch** [comment]
Wait! You did actually remove the two empty lines in your first commit.

**Comment by ddkn** [comment]
Oh I see what you mean, there are two lines above and below that got removed and should have two spaces? Re-add them? or should I submit a new PR?

**Comment by laloch** [comment]
Nope, just add them back and force-push your topic branch. You can do anything you want with the topic branches here on GitHub. The open PRs will get automatically updated with the new content.

**Comment by ddkn** [comment]
Okay looks like everything worked now. Thanks again for the help!

**Comment by laloch** [comment]
You're welcome. Your contribution is very much appreciated.

File: xonsh/aliases.py [hunk 1]
```diff
@@ -820,6 +821,11 @@ def sudo(args):
         default_aliases["grep"] = ["grep", "--color=auto"]
         default_aliases["egrep"] = ["egrep", "--color=auto"]
         default_aliases["fgrep"] = ["fgrep", "--color=auto"]
+    elif ON_OPENBSD:
+        default_aliases["grep"] = ["grep"]
+        default_aliases["egrep"] = ["egrep"]
+        default_aliases["fgrep"] = ["fgrep"]
+        default_aliases["ls"] = ["ls"]
```
Comment (laloch) [inline xonsh/aliases.py]: Why do you introduce aliases which do not modify their respective command in any way?
If you don't want any aliases on OpenBSD, just say `pass` here.

Comment (laloch) [reply xonsh/aliases.py]: Thanks!

**Comment by laloch** [comment]
@xonsh/xore, I think we could merge this without extensive testing. It seems that we have exactly one OpenBSD user and he made things work for himself. Any objections?

**Comment by gforsyth** [comment]
And thanks for putting this in @ddkn!!

**Pull request merged**